### PR TITLE
Updates for PNC

### PR DESCRIPTION
**Pull request by SDawley**
Updated with the changes necessary to build via Project Newcastle, so the artifacts can then be pulled into Brew. 
Added a gradle.properties file with a group and version, which are both necessary for PNC alignment, though either can be changed.
Added the maven-publish plugin so artifacts can be published to the PNC repo.

### CODE REVIEW

**Comment by SDawley** [comment]
@nickboldt @svor Sanity checks welcome.

File: server/build.gradle [hunk 1]
```diff
@@ -143,6 +144,18 @@ jooq {
     }
 }
 
+publishing {
+    publications {
+        maven(MavenPublication) {
+            //artifact ('build/libs/*.jar')
```
Comment (svor) [inline server/build.gradle:150]: do we need this?

File: server/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+version=1.0.0
+daemon=false
+group=eclipse.che
```
Comment (svor) [inline server/gradle.properties:3]: please add empty line

**Review by svor — CHANGES_REQUESTED** [review]
DS uses [che-openvsx branch](https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/build/dockerfiles/openvsx-builder.Dockerfile#L18) to build openvsx server.
You have to create a PR into **che-openvsx** branch instead of **master**

File: server/gradle.properties (hunk 2 above)
Comment (nickboldt) [inline server/gradle.properties:3]:
```suggestion
group=eclipse.che

```

**Review by nickboldt — APPROVED** [review]
beyond @svor's comment, this looks good to me but also I have no idea what's happening here :) 

I didn't realize you'd be connecting this build to PNC artifacts... is this to use extensions in UPSTREAM builds? If so, um... PNC is behind the RH firewall so I don't know that you'll be able to access artifacts via a GH action. 

Maybe I missed something but I thought the purpose of building built-in and plugin reg vsix files was ONLY for Dev Spaces, not for Che's openvsx instance as well.

**Comment by SDawley** [comment]
The publish is needed _for PNC_. PNC does not capture artifacts from the built itself, it requires the build to include a publish, which the PNC alignment stage configures to publish to the repo behind the Red Hat firewall. Since the build uses this repo as source (unless there's a downstream DS-openvsx that I'm not aware of, which is possible) I made the changes here.

**Review by svor — APPROVED** [review]
LGTM

**Comment by SDawley** [comment]
It looks like I cannot merge could someone merge or add me?

**Review by svor — CHANGES_REQUESTED** [review]
@SDawley 
I've tried to apply your changes in [openvsx-builder.Dockerfile](https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/build/dockerfiles/openvsx-builder.Dockerfile#L16-L18) :
```
RUN cd /tmp && \
    git clone https://github.com/SDawley/che-openvsx.git && \
    cd che-openvsx/server && \
    git checkout PNC
```
and the plugin registry build was failed:
```
#9 [ 5/10] RUN mkdir /openvsx-server &&     cp /tmp/che-openvsx/server/scripts/run-server.sh /openvsx-server &&     cp /tmp/che-openvsx/server/build/libs/openvsx-server.jar /openvsx-server
#9 0.495 cp: cannot stat '/tmp/che-openvsx/server/build/libs/openvsx-server.jar': No such file or directory
#9 ERROR: executor failed running [/bin/sh -c mkdir /openvsx-server &&     cp /tmp/che-openvsx/server/scripts/run-server.sh /openvsx-server &&     cp /tmp/che-openvsx/server/build/libs/openvsx-server.jar /openvsx-server]: exit code: 1
------
 > [ 5/10] RUN mkdir /openvsx-server &&     cp /tmp/che-openvsx/server/scripts/run-server.sh /openvsx-server &&     cp /tmp/che-openvsx/server/build/libs/openvsx-server.jar /openvsx-server:
#9 0.495 cp: cannot stat '/tmp/che-openvsx/server/build/libs/openvsx-server.jar': No such file or directory
------
error: failed to solve: executor failed running [/bin/sh -c mkdir /openvsx-server &&     cp /tmp/che-openvsx/server/scripts/run-server.sh /openvsx-server &&     cp /tmp/che-openvsx/server/build/libs/openvsx-server.jar /openvsx-server]: exit code: 1
```
is it expected?

**Review by svor — APPROVED** [review]
lgtm, it builds now
@SDawley you should have an access to merge the PR now

**Pull request merged**